.github/workflows/docs-content-check-legacy-links-format.yml

```
@@ -0,0 +1,123 @@
+              repo: process.env.REPO,
+              commit_sha: process.env.COMMIT_SHA
+            })
+            return data[0].number
+
+      - name: Get list of PR files
+        id: get-lists-of-pr-files
+        uses: actions/github-script@v6
+        env:
+          PR_NUMBER: ${{ steps.get-pr-number.outputs.result }}
+          MDX_DIRECTORY: ${{ inputs.mdx-directory }}
+          NAV_DATA_DIRECTORY: ${{ inputs.nav-data-directory }}
+        with:
+          script: |
+            const changedMdxFiles = []
+            const changedNavDataJsonFiles = []
+
+            // A helper for recursively fetching all files for a PR
+            const fetchAllPrFiles = async ({ allPrFiles = [], page = 1 }) => {
+              const { data } = await github.rest.pulls.listFiles({
+                owner: process.env.OWNER,
+                repo: process.env.REPO,
+                per_page: 100,
+                pull_number: process.env.PR_NUMBER,
+                page,
+              })
+              data.forEach(file => {
+                allPrFiles.push(file)
+              })
+
+              if (data.length === 100) {
+                await fetchAllPrFiles({ allPrFiles, page: page + 1 })
+              }
+            }
+
+            // Fetch all the PR files
+            const allPrFiles = []
+            await fetchAllPrFiles({ allPrFiles })
+
+            // Sort the PR files for what we're interested in
+            allPrFiles.forEach(file => {
+              if (file.status === 'removed') {
+                return
+              }
+              if (file.filename.endsWith('.mdx')) {
+                const filenameWithoutDirectory = file.filename.replace(process.env.MDX_DIRECTORY, '')
+                changedMdxFiles.push(filenameWithoutDirectory)
+                return
+              }
+              // @TODO uncomment when new nav-data formats can be considered
+              // valid by validateNode() in react-components/docs-sidenav
+              //if (file.filename.endsWith('-nav-data.json')) {
+              //  const filenameWithoutDirectory = file.filename.replace(process.env.NAV_DATA_DIRECTORY, '')
+              //  changedNavDataJsonFiles.push(filenameWithoutDirectory)
+              //}
+            })
+
+            // Log and return the result
+            const result = { changedMdxFiles, changedNavDataJsonFiles }
+            console.log('get-lists-of-pr-files result:', result)
+            return result
+
+      - name: Checkout dev-portal
+        uses: actions/checkout@v3
+        with:
+          repository: 'hashicorp/dev-portal'
+          ref: 'amb.explore-docs-content-links'
+
+      - name: Install dev-portal dependencies
+        run: npm ci
+
+      - name: Check out branch with content changes
+        uses: actions/checkout@v3
+        with:
+          path: 'content-repo'
+          repository: '${{ inputs.repo-owner }}/${{ inputs.repo-name }}'
+          ref: 'docs/amb.migrate-link-formats'
```

TODO remove the `ref`s in both of these places right before merging. They were only needed in order to test that this workflow is successful and can be called from another repo.

Waypoint testing happened in these runs:
https://github.com/hashicorp/waypoint/actions/workflows/check-legacy-links-format.yml

```suggestion
      - name: Checkout dev-portal
        uses: actions/checkout@v3
        with:
          repository: 'hashicorp/dev-portal'

      - name: Install dev-portal dependencies
        run: npm ci

      - name: Check out branch with content changes
        uses: actions/checkout@v3
        with:
          path: 'content-repo'
          repository: '${{ inputs.repo-owner }}/${{ inputs.repo-name }}'
```